test/common/crudtester.go

```diff
@@ -739,3 +731,25 @@ func (c *FederatedTypeCrudTester) CheckStatusCreated(qualifiedName util.Qualifie
 		c.tl.Fatalf("Timed out waiting for %s %q", statusKind, qualifiedName)
 	}
 }
+
+// GetGenericStatus retrieves a federated resource and converts it to
+// the generic status interface.
+func GetGenericStatus(client genericclient.Client, gvk schema.GroupVersionKind,
+	qualifiedName util.QualifiedName) (*status.GenericFederatedStatus, error) {
+
+	fedObject := &unstructured.Unstructured{}
+	fedObject.SetGroupVersionKind(gvk)
+	err := client.Get(context.TODO(), fedObject, qualifiedName.Namespace, qualifiedName.Name)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to retrieve federated resource from the API")
+	}
+
+	// Convert the resource to the status interface
+	genericStatus := &status.GenericFederatedStatus{}
+	err = util.UnstructuredToInterface(fedObject, genericStatus)
```

Comment:
How does `fedObject`, which is the unstructured version of a `FederatedX` resource, get unmarshalled to a `GenericFederatedStatus` struct? Don't we need to be working with the `status` field from the unstructured version of `FederatedX` and the `GenericFederatedStatus` struct?

Reply:
Take a look at `GenericFederatedStatus` - it has an optional `Status` field.  Unmarshalling from the unstructured resource to the struct will populate the `Status` field in the struct if it is present in the resource.